### PR TITLE
Add strike

### PR DESCRIPTION
**Pull request by CloudyMcFox**

### CODE REVIEW

File: command_handler.py [hunk 1]
```diff
@@ -35,7 +36,9 @@ def __init__(self):
       'magic8ball': ( magic8ball.Magic8Ball(), None ),
       'question': ( questions.Question(), "answers your questions"),
       'badbot': ( badbot.BadBot(), None ),
-      'goodbot': ( badbot.BadBot(), None )
+      'goodbot': ( badbot.BadBot(), None ),
+      'strike': ( strike.Strike(), None),
```
Comment (uncanardnoir) [inline command_handler.py]: this is equivalent to creating 2 instances of strike class which is fine if it's stateless but better design to create 1 and pass a pointer to both -- see the meme ones

Comment (uncanardnoir) [reply command_handler.py]: i just saw i'm doing the same with badbot :(

File: handlers/strike.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import os
+from jsonpath_rw import jsonpath, parse
+
+class Strike():
+	def handleCommand(self, command):
+
```
Comment (uncanardnoir) [inline handlers/strike.py]: extra whitespace

File: handlers/strike.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import os
+from jsonpath_rw import jsonpath, parse
+
+class Strike():
+	def handleCommand(self, command):
+	
+		if command.command == "getStrikes"
+			return getStrikesString(self)
+		userId = parse('$.message.from.id').find(command.parent)
```
Comment (uncanardnoir) [inline handlers/strike.py]: return self.getStrikesString()

File: handlers/strike.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import os
+from jsonpath_rw import jsonpath, parse
+
+class Strike():
+	def handleCommand(self, command):
+	
+		if command.command == "getStrikes"
+			return getStrikesString(self)
```
Comment (uncanardnoir) [inline handlers/strike.py]: we use 2 spaces for an indent level in this proj... mind doing ctrl-h find replace?

File: command_handler.py (hunk 1 above)
Comment (CloudyMcFox) [reply command_handler.py]: Done.

File: handlers/strike.py (hunk 3 above)
Comment (CloudyMcFox) [reply handlers/strike.py]: Fixed

File: handlers/strike.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+		  with open(userFile, "w") as f:
+			f.write("{0}\n1".format(strikeUser))
+		else:
+		  with open(userFile, "r+") as f:
+			data = f.read().split()
+			nStrikes = int(data[1]) + 1
+			f.seek(0)
+			f.write("{0}\n{1}".format(strikeUser, nStrikes))
+			f.truncate()
+		return "Careful {0}. You are now at {1} strikes\n".format(strikeUser, nStrikes)
```
Comment (uncanardnoir) [inline handlers/strike.py]: don't need the \n at end of line (i think that will just make the message bigger in telegram?)

File: handlers/strike.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	
+		if command.command == "getStrikes"
+			return getStrikesString(self)
+		userId = parse('$.message.from.id').find(command.parent)
+		strikeUser = command.args
+		if not userId or not strikeUser:
+			return None
+		userId = userId[0].value
+		userFile = 'strike/{0}'.format(strikeUser)
+		if not userId == 313082320 #Ryan's id
```
Comment (just-park) [inline handlers/strike.py]: bug

File: handlers/strike.py (hunk 4 above)
Comment (CloudyMcFox) [reply handlers/strike.py]: notepad++ :(

Comment (CloudyMcFox) [reply handlers/strike.py]: done

File: handlers/strike.py (hunk 5 above)
Comment (CloudyMcFox) [reply handlers/strike.py]: Done

File: handlers/strike.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+from jsonpath_rw import jsonpath, parse
+
+class Strike():
+	def handleCommand(self, command):
+	
+		if command.command == "getStrikes"
+			return getStrikesString(self)
+		userId = parse('$.message.from.id').find(command.parent)
+		strikeUser = command.args
+		if not userId or not strikeUser:
```
Comment (just-park) [inline handlers/strike.py]: please validate the user is in the channel.
also if not found return that string.

Comment (CloudyMcFox) [reply handlers/strike.py]: Yeah thats next, a lot harder to do :( pending

File: handlers/strike.py (hunk 6 above)
Comment (CloudyMcFox) [reply handlers/strike.py]: NO.

File: handlers/strike.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      f.write("{0}\n{1}".format(strikeUser, nStrikes))
+      f.truncate()
+    return "Careful {0}. You are now at {1} strikes.".format(strikeUser, nStrikes)
+
+  def getStrikesString(self):
+    ret = ''
+    nStrikes = 0
+    for filepath in os.listdir('strike'):
+      with open(os.path.join('strike', filepath), "r") as f:
+      data = f.read().split()
```
Comment (uncanardnoir) [inline handlers/strike.py:33]: indentation levels

File: handlers/strike.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      thisStrike = int(data[1])
+      thisName = data[0]
+      thisline = '{0}: {1} strike{2}\n'.format(thisName, thisStrike, 's' if thisStrike != 1 else '')
+      ret += thisline
```
Comment (uncanardnoir) [inline handlers/strike.py:37]: did you try running this? i thought += isn't allowed in python?

**Review by uncanardnoir — APPROVED** [review]

**Pull request merged**